src/datautils/wellboreItemShapeGenerator.ts

```diff
@@ -1,78 +1,116 @@
+export const findCasing = (id: string, casings: Casing[]): Casing => {
+  const res = casings.filter((c) => c.casingId === id);
+  if (res.length === 0) {
+    throw new Error('Casing not found');
   }
+  return res[0];
+};
 
-  const tension = 0.2;
-  const numPoints = 999;
-  const normalOffsetCoordsUpInterpolator = new CurveInterpolator(normalOffsetCoordsUpOrig.map(pointToArray), tension);
-  const normalOffsetCoordsDownInterpolator = new CurveInterpolator(normalOffsetCoordsDownOrig.map(pointToArray), tension);
-  const normalOffsetCoordsUp = normalOffsetCoordsUpInterpolator.getPoints(numPoints).map(arrayToPoint);
-  const normalOffsetCoordsDown = normalOffsetCoordsDownInterpolator.getPoints(numPoints).map(arrayToPoint);
+export const overlaps = (top1: number, bottom1: number, top2: number, bottom2: number): boolean => top1 <= bottom2 && top2 <= bottom1;
 
-  return { wellBorePathCoords, normalOffsetCoordsDown, normalOffsetCoordsUp };
-};
+export const uniq = <T>(arr: T[]): T[] => Array.from<T>(new Set(arr));
 
-export const actualPoints = (s: HoleObjectData): Point[] => {
-  let start = new Point();
-  let stop = new Point();
-  let startIndex = 0;
-  let stopIndex = 0;
-  const a = s.points.filter((p: MDPoint, index: number) => {
-    if (s.data.start > p.md) {
-      startIndex = index;
-    }
-    if (s.data.end >= p.md) {
-      stopIndex = index;
-    }
-    return p.md > s.data.start && p.md < s.data.end;
-  });
-
-  if (a == null || a.length === 0) {
-    return [];
-  }
+export const findIntersectingItems = (
+  cement: Cement,
+  parentCasing: Casing,
+  casings: Casing[],
+  holes: HoleSize[],
+): { holes: HoleSize[]; casings: Casing[] } => {
+  const { toc: start } = cement;
+  const { end } = parentCasing;
+  const overlappingHoles = holes.filter((h: HoleSize) => overlaps(start, end, h.start, h.end) && h.diameter > parentCasing.diameter);
+  const overlappingCasings = casings.filter(
+    (c: Casing) => c.casingId !== cement.casingId && overlaps(start, end, c.start, c.end) && c.diameter > parentCasing.diameter,
+  );
 
-  startIndex -= 0;
-  stopIndex += 0;
-  start = s.points[startIndex >= 0 ? startIndex : 0].point;
-  stop = s.points[stopIndex <= s.points.length ? stopIndex : s.points.length - 1].point;
-  return [start, ...a.map((b: MDPoint) => b.point), stop];
+  return {
+    holes: overlappingHoles,
+    casings: overlappingCasings,
+  };
 };
 
-export const findCasing = (id: string, casings: any) => {
-  const res = casings.filter((c: any) => c.casingId === id);
-  return res.length > 0 ? res[0] : {};
+export const parseCement = (cement: Cement, casings: Casing[], holes: HoleSize[]): CompiledCement => {
+  const attachedCasing = findCasing(cement.casingId, casings);
+  const res: CompiledCement = {
+    ...cement,
+    boc: attachedCasing.end,
+    attachedCasing,
+    intersectingItems: findIntersectingItems(cement, attachedCasing, casings, holes),
+  };
+  return res;
```

Comment:
This name is a bit unfortunate, it is not very obvious what this function is doing. The only clear thing is that it is creating a `CompiledCement` out of a `Cement`, so maybe let's name it `compileCement` or something along those lines